src/_pytest/python.py

```diff
@@ -911,6 +911,124 @@ def hasnew(obj: object) -> bool:
     return False
 
 
+@final
+@attr.s(frozen=True, auto_attribs=True)
+class IdMaker:
+    argnames: Iterable[str] = attr.Factory(list)
```

Comment:
When refactoring to a class, it's less natural to assume it will only be called once. So I think it will be better to avoid `Iterable` and use `Sequence` instead.

Reply:
> less natural to assume it will only be called once

Yes... It shouldn't be a problem to iterate over an `Iterable` multiple times (`__iter__()` will provide a fresh iterator on each call). But users of `IdMaker` may expect index access to `argnames` and `parametersets`, so `Sequence` is the better choice anyway.